src/nectar/transactionbuilder.py

```
@@ -398,24 +427,45 @@ def get_block_params(self, use_head_block=False):
-        if self.blockchain.use_sc2:
-            return
+        if self.blockchain.use_hs and self.blockchain.hivesigner is not None:
+            # Use HiveSigner for signing
+            try:
+                signed_tx = self.blockchain.hivesigner.sign(self.json())
+                if signed_tx and "signatures" in signed_tx:
+                    # Attach the signatures from HiveSigner to the transaction
+                    self["signatures"] = signed_tx["signatures"]
+                    return signed_tx
+                else:
+                    raise ValueError("HiveSigner failed to sign transaction")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)